scene/resources/packed_scene.cpp

```diff
@@ -304,6 +304,13 @@ Node *SceneState::instantiate(GenEditState p_edit_state) const {
 
 					ERR_FAIL_INDEX_V(nprops[j].name, sname_count, nullptr);
 
+					// If there's a custom type script present, make this script it's baseline
+					if (snames[nprops[j].name] == CoreStringNames::get_singleton()->_custom_type_script) {
+						if (!props[nprops[j].value].is_null()) {
+							node->set(CoreStringNames::get_singleton()->_script, props[nprops[j].value], &valid);
```

Comment:
Doesn't it need the same treatment as `script` below?
Though I don't fully get the purpose of this code 🤔

Reply:
When a packed scene is loaded, `custom_type_script` gets loaded first. If it exists it preemptively sets `script` to its value in case the packed scene has no `script` value stored in it (remember the case you mentioned where storing `script` might not be necessary?)

I think I added this before making sure `script` was always stored together with `custom_type_script` inside packed scenes.